### ISSUE TITLE
feedback on hardcoded python names

### ISSUE THREAD

**Issue by ghost** [rtbo/rust-xcb#62]
Hi, great project!

Here's two small changes I had to apply while building a software in NetBSD vanilla (not using pkgsrc):

1. in rs_client.py you assume that "python3" exists. There are plenty of systems where python3 does not exist or is a symlink to whatever the python version is right now. Not hardcoding this would be the most obvious choice.
2. in build.rs I had to apply the same change (python3 -> python3.7).

I've dealt with this before, but in autotools. The path of least resistance would be to take a list of possible python executable names in $PATH to check (only names, not paths) and use it in the python file. The rust file is probably more flexible, using an array should be possible.. or both could use an file to read from..

Without this, it's not possible to build for example alacritty on NetBSD.

**Comment by rtbo** [comment]
The shebang line in the python file is of seldom use as it is normally invoked by the rust build script, but if you wish to fix it, we'd have to specify more than one executable. Is it feasible?
The rust script itself should be straightforward to fix.
What executable names would you recommend for searching?

**Comment by ng-0** [comment]
With regards to the hashbang:
I think simply patching it on an as-needed base might be the best solution, which is what I do right now. However this makes it necessary to have an python3 symlink in your path.
It seems like I did not remember the content of PEP 394 correctly (https://www.python.org/dev/peps/pep-0394/). I thought it explicitly stated that no "python3" executable must exist.
It seems as if my conclusion in the buildsystem for gnunet was right, python3 as well as python3.x to python3.8 (etc) can exist. I have simply forgotten that some systems really distribute an python3 named executable.

Changing the hashbang would alter the checksum of the file. The build.rs is invoked by the build.rs. Is the file called in a way which would make the hashbang unnecessary (for example python3 $file)? If it could simply be removed and the python file is only called during build, this seems like the best approach to deal with the hashbang and executable names changing.

With regards to the executable names in the build.rs:
The list (according to the PEP I linked above) usually goes like python3, python3.0, python3.1, ..., python3.7, python3.x
where .x is the current final version of python3.

What I (have to) do in autotools is this:
https://git.gnunet.org/gnunet.git/tree/configure.ac#n227
which, as far as my PEP reading goes is PEP compliant and matches what distributors do out there.
